fix(shutdown): graceful shutdown times out after 10s forcing exit

## Observation\n\nServer received SIGTERM at 10:18:37, attempted graceful shutdown:\n\n```\n[INFO] Received SIGTERM, shutting down gracefully (timeout=10s)...\n[INFO] Sent shutdown notification to 3 SSE clients\n[INFO] Cancelling orchestrator...\n[INFO] Closed 3 SSE clients\n[INFO] Pulse stopped after 7 beats / 25 beats\n[INFO] stats: h2=0 h1=230\n[INFO] Closed 5 SSE connections\n[INFO] Shutdown complete.\n[INFO] Evicted 8 expired SSE buffer events\n[INFO] reaped 0 SSE guards + 0 HTTP guards + 5 stale sessions\n[ERROR] Graceful shutdown timed out after 10s, forcing exit.\n```\n\n## Analysis\n\n- `Shutdown complete` logs before the timeout, yet the process still doesn't exit.\n- The post-shutdown eviction and reap tasks fire after `Shutdown complete`, suggesting some fiber or background loop is still alive after the orchestrator cancellation.\n- Pulse stopped normally (7 and 25 beats), SSE clients closed, but something else holds the Eio switch open.\n\n## Impact\n\nForced exit may skip final cleanup (file flushes, state persistence). On restart the launcher detects the port free and boots a new instance, so availability is not affected.\n\n## Suggested Investigation\n\n- Identify which fiber survives past `Shutdown complete`\n- The SSE eviction timer and session reaper appear to run after shutdown — check if they are attached to the main switch

--- a/bin/main_eio.ml
+++ b/bin/main_eio.ml
@@ -233,9 +233,16 @@ let make_extended_handler routes =
       Http.Response.internal_error msg reqd
 
 (** Main server loop *)
-let run_server ~sw ~env ~host ~port ~base_path =
+let run_server ~sw:_ ~env ~host ~port ~base_path =
+  (* Use a dedicated sub-switch so that ALL fibers spawned by
+     Server_runtime_bootstrap (background maintenance, keeper loops,
+     dashboard refresh, etc.) are children of this switch.  When
+     Eio.Fiber.first cancels the run_server fiber on SIGTERM, the
+     sub-switch is cancelled too, which propagates Cancel to every
+     child fiber — preventing the 10s force-exit timeout. *)
+  Eio.Switch.run @@ fun server_sw ->
   try
-    Server_runtime_bootstrap.run ~sw ~env ~host ~port ~base_path ~make_routes
+    Server_runtime_bootstrap.run ~sw:server_sw ~env ~host ~port ~base_path ~make_routes
       ~make_request_handler:make_extended_handler
       ~make_h2_request_handler:Server_h2_gateway.make_request_handler
       ~make_h2_error_handler:Server_h2_gateway.make_error_handler
@@ -313,7 +320,6 @@ let run_cmd host port base_path =
   let rec try_start attempt =
     (try
       Eio.Switch.run @@ fun sw ->
-      Masc_mcp.Dashboard_cache.set_sw sw;
       let clock = Eio.Stdenv.clock env in
       let rec await_shutdown_signal () =
         match Atomic.exchange pending_shutdown_signal None with

--- a/lib/server/server_runtime_bootstrap.ml
+++ b/lib/server/server_runtime_bootstrap.ml
@@ -30,6 +30,7 @@ let create_server_state ~sw ~base_path ~clock ~mono_clock ~net ~proc_mgr ~fs
   Mcp_eio.set_net net;
   Mcp_eio.set_clock clock;
   Eio_context.set_switch sw;
+  Dashboard_cache.set_sw sw;
   Eio_context.set_net net;
   Eio_context.set_clock clock;
   Eio_context.set_mono_clock mono_clock;

